sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzMonListExtensions.cs

```diff
@@ -220,12 +220,85 @@ internal static string GetHostUsingNetPeerAttributes(this AzMonList tagObjects)
             return host;
         }
 
+        ///<summary>
+        /// Gets Http dependency target from activity tag objects.
+        ///</summary>
+        internal static string GetHttpDependencyTarget(this AzMonList tagObjects)
+        {
+            string target;
+            string defaultPort = GetDefaultHttpPort(AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpScheme)?.ToString());
+            var peerService = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributePeerService)?.ToString();
+            if (!string.IsNullOrWhiteSpace(peerService))
+            {
+                target = peerService;
+                return target;
+            }
+
+            var httpHost = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpHost)?.ToString();
+            if (!string.IsNullOrWhiteSpace(httpHost))
+            {
+                string portSection = $":{defaultPort}";
+                if (httpHost.EndsWith(portSection, StringComparison.OrdinalIgnoreCase))
+                {
+                    var truncatedHost = httpHost.Substring(0, httpHost.IndexOf(portSection, StringComparison.OrdinalIgnoreCase));
+                    target = truncatedHost;
+                }
+                else
+                {
+                    target = httpHost;
+                }
+                return target;
+            }
+
+            var httpUrl = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpUrl)?.ToString();
+            if (!string.IsNullOrWhiteSpace(httpUrl) && Uri.TryCreate(httpUrl.ToString(), UriKind.RelativeOrAbsolute, out var uri) && uri.IsAbsoluteUri)
+            {
+                target = uri.Authority;
+                if (!string.IsNullOrWhiteSpace(target))
+                {
+                    return target;
+                }
+            }
+
+            target = tagObjects.GetHostUsingNetPeerAttributes();
+            if (!string.IsNullOrWhiteSpace(target))
```

Comment:
This part is repeating in both the places, should we have agressiveinline method to extract this part out.

Reply:
Done!